packages/compile-solidity/src/compilerSupplier/index.ts

```diff
@@ -0,0 +1,117 @@
+import path from "path";
+import fs from "fs";
+import semver from "semver";
+
+import { Docker, Local, Native, VersionRange } from "./loadingStrategies";
+
+const defaultSolcVersion = "0.5.16";
+
+export class CompilerSupplier {
+  private events: any;
+  private version: string;
+  private docker: boolean;
+  private compilerRoots: string[];
+  private strategyOptions: Partial<{
+    version: string;
+    docker: boolean;
+    compilerRoots: string[];
+    events: any; // represents a @truffle/events instance, which lacks types
+    spawn: {
+      maxBuffer: number;
+    };
+  }>;
+
+  constructor({ events, solcConfig }) {
+    const { version, docker, compilerRoots, spawn } = solcConfig;
+    this.events = events;
+    this.version = version ? version : defaultSolcVersion;
+    this.docker = docker;
+    this.compilerRoots = compilerRoots;
+    this.strategyOptions = {};
+    if (version) this.strategyOptions.version = this.version;
+    if (docker) this.strategyOptions.docker = compilerRoots;
```

Comment:
@haltman-at writes [here](https://github.com/trufflesuite/truffle/pull/4222#pullrequestreview-728040585) that this is likely a mistake.

Reply:
(Pushed fix)

Reply:
I'm not seeing the fix?

Reply:
Hm, this raises another issue though -- like `spawn` and `compilerRoots`, the `dockerTagsUrl` option is undocumented.  I'll go comment on the existing issue to note that.